### ISSUE TITLE
 Apply KPConv on an large ALS data set

### ISSUE THREAD

**Issue by mnslarcher** [HuguesTHOMAS/KPConv-PyTorch#11]
Hi Thomas, 

First of all, congratulations for your work and thank you in advance if you find time to answer my questions.

I work for an electric utility company, next month I will be part of a project whose goal is to classify a large ALS point cloud (~60,000 km already human labeled). I have some experience in computer vision but this will be my first project on this topic, so forgive me if I ask naive questions.

* In your opinion, is KPConv a good choice for this type of data set? I say yes given [this](https://npm3d.fr/paris-lille-3d) and [this](https://arxiv.org/pdf/2004.11985.pdf) promising results, but if you know any reason why is not, I would be very interested to know it
* What are the main steps that you suggest to take to adapt your scripts to a different data set (in particular to a large ALS data set where one of the most important classes is the powerline)? 
* What are the main hyper-parameters which need to be tuned when changing the data set? I imagine for sure the input sphere radius and dl_0
* Does this implementation support multi-GPUs training?

Any further feedback is appreciated, I have read your paper but I still need to study your repository, I would like first to have a confirmation that I am looking in the right direction before investing time to study more your excellent work :)

**Comment by HuguesTHOMAS** [comment]
Hi Mario,

Thanks for your interest in my work. I definitely think KPConv is a good solution to classify ALS point clouds. Their might be some more specific method that could achieve a few percent better that KPConv in the case of ALS data, but if you consider generic point cloud segmentation algorithms, KPConv is among the best methods today.

To adapt the code to a new dataset, you have to write a new dataset script in the `datasets` folder. In your case, the easiest way to do it is to reproduce the `S3DIS.py` dataset script, and change the parameters so that it works on your own data. Follow the code execution from `train_S3DIS.py` and adapt the function that loads data.

About the parameters, you are perfectly right, the input radius and dl0 are the two most important parameters when changing the nature of the data. in you case you want to increase them. When you will be able to run the code, you can play with these parameters, but remember the ratio between them controls the size of what the network "sees". You can see details on the effect of these parameters in previous issues https://github.com/HuguesTHOMAS/KPConv-PyTorch/issues/5 and https://github.com/HuguesTHOMAS/KPConv-PyTorch/issues/2

Eventually, the implementation does not support multi GPU training. I did not really look into it , because I don't need it, but it might be simple to do with PyTorch, I have no idea.

Good luck with your project. If you have more questions, I'll be happy to help.

**Comment by mnslarcher** [comment]
Hi @HuguesTHOMAS, thanks for your quick and very useful answers :). I will definitely follow your suggestions and study your repo further. 

If I'll extend this implementation to run with multi-gpu (it shouldn't be very difficult) I will inform you in the event that one day you or other users of the repository will need this functionality.

Just a couple of additional questions before digging deeper into this particular implementation.

Are the results obtained with the original repository completely achievable with this? I would prefer to use PyTorch but if there are strong reasons to use the TensorFlow implementation or the other PyTorch implementation that you linked in your README I'll consider them. 

Have you noticed any substantial differences in terms of time or accuracy?

Thanks

**Comment by HuguesTHOMAS** [comment]
I would say this PyTorch implementation is the best you can use right now. It achieves the same performances as my old Tensorflow repo, and is up-to-date because I wrote it more recently. The only drawback would be that you have more chances to encounter a bug as it has not been tested as much as the tensorflow one, but I try to be reactive and help with that.

**Comment by mnslarcher** [comment]
Great, then I'll stick with this one :)! Thanks
